libs/coin-modules/coin-bitcoin/src/wallet-btc/__tests__/wallet.storage.unit.test.ts

```diff
@@ -378,4 +378,285 @@ describe("Unit tests for bitcoin storage", () => {
+      index: 0,
+    });
+    expect(unspentUtxos).toHaveLength(0);
+  });
+
+  it("[utxos] should not allow selecting spent UTXOs for new transactions", () => {
+    // Setup: append a spent UTXO
+    storage.appendTxs([
+      {
+        id: "tx-spent",
+        inputs: [],
+        outputs: [
+          {
+            output_index: 0,
+            value: "100000",
+            address: "test-address",
+            output_hash: "tx-spent",
+            block_height: 110,
+            rbf: false,
+          },
+        ],
+        block: { hash: "block-spent", height: 110, time: new Date().toISOString() },
+        account: 0,
+        index: 0,
+        address: "test-address",
+        received_at: new Date().toISOString(),
+      },
+      {
+        id: "tx-spending",
+        inputs: [
+          {
+            output_hash: "tx-spent",
+            output_index: 0,
+            value: "100000",
+            address: "test-address",
+            sequence: 4294967294,
+          },
+        ],
+        outputs: [],
+        block: { hash: "block-spending", height: 111, time: new Date().toISOString() },
+        account: 0,
+        index: 0,
+        address: "test-address",
+        received_at: new Date().toISOString(),
+      },
+    ]);
+
+    // Verify spent UTXO is not available
+    const availableUtxos = storage.getAddressUnspentUtxos({
+      address: "test-address",
+      account: 0,
+      index: 0,
+    });
+    expect(availableUtxos).toHaveLength(0);
+  });
+
+  it("[utxos] should prevent race condition issues on UTXO updates", () => {
+    storage.appendTxs([
+      {
+        id: "tx-race",
+        inputs: [],
+        outputs: [
+          {
+            output_index: 0,
+            value: "100000",
+            address: "race-address",
+            output_hash: "tx-race",
+            block_height: 200,
+            rbf: false,
+          },
+        ],
+        block: { hash: "block-race", height: 200, time: new Date().toISOString() },
+        account: 0,
+        index: 0,
+        address: "race-address",
+        received_at: new Date().toISOString(),
+      },
+    ]);
+
+    // Simulate rapid conflicting updates
+    storage.removeTxs({ account: 0, index: 0 });
+
+    storage.appendTxs([
+      {
+        id: "tx-race",
+        inputs: [],
+        outputs: [
+          {
+            output_index: 0,
+            value: "100000",
+            address: "race-address",
+            output_hash: "tx-race",
+            block_height: 200,
+            rbf: false,
+          },
+        ],
+        block: { hash: "block-race", height: 200, time: new Date().toISOString() },
+        account: 0,
+        index: 0,
+        address: "race-address",
+        received_at: new Date().toISOString(),
+      },
+    ]);
+
+    const utxos = storage.getAddressUnspentUtxos({ address: "race-address", account: 0, index: 0 });
+    expect(utxos).toHaveLength(1);
+    expect(utxos[0].output_hash).toEqual("tx-race");
```

Comment:
```suggestion
    expect(utxos).toEqual([
      expect.objectContaining({
        output_hash: "tx-race",
      }),
    ]);
```
All your assertions could be combined